src/renderer/components/MapEditor/ICCAExportDialog.js

```diff
@@ -0,0 +1,494 @@
+  aboutYouSection: 'About you:',
+  // Question prompts
+  question1Prompt: `
+    Has your community or indigenous people collectively consented to the map
+    and other information being shared with UNEP-WCMC?`,
+  question2Prompt: `
+    Do you want the map and other information to be available for anyone to
+    view and download?`,
+  question3Prompt: `
+    Name of the community, indigenous people, or individual providing the
+    information`,
+  question4Prompt: `
+    If you are an individual, what is your relationship to the indigenous
+    people or community?`,
+  question5Prompt: `
+    Email address, phone number or address where we can contact you. We will
+    contact you to confirm we have received your information and to discuss
+    the information further where needed. We will also contact you periodically
+    to make sure the information is up to date.`,
+  // Yes/no answers
+  answerYes: 'Yes',
+  answerNo: 'No',
+  // Choices for question 4
+  question4Answer1: 'Member of the community/indigenous people',
+  question4Answer2: 'Representative or associate of the community/indigenous people',
+  question4Answer3: 'Representative of a non-governmental organisation',
+  question4Answer4: 'Other',
+  // Placeholder for question 5
+  question5Placeholder: 'Your contact information',
+  // Helper texts
+  requiredAnswer: 'This answer is required.',
+  atLeastOneAnswer: 'Please fill in at least one field.',
+  errorMissingRequired: 'Please fill out all required fields.',
+  // Labels for community name
+  communityOriginalName: 'Original language',
+  communityEnglishName: 'English'
+})
+
+// Insert a tiny lil' shake animation when an error occurs
+// We don't use the makeStyles api here because it can't handle keyframes
+insertCss(`
+  @keyframes shakeX {
+    from,
+    to {
+      transform: translate3d(0, 0, 0);
+    }
+
+    16.67%,
+    50%,
+    83.33% {
+      transform: translate3d(-10px, 0, 0);
+    }
+
+    33.33%,
+    66.67% {
+      transform: translate3d(10px, 0, 0);
+    }
+  }
+
+  .shakeX {
+    animation-name: shakeX;
+    animation-timing-function: ease-in-out;
+    animation-duration: 500ms;
+    animation-delay: 0;
+  }
+`)
+
+const EditDialogContent = ({ onClose }) => {
+  const classes = useStyles()
+  const { formatMessage } = useIntl()
+  const [saving, setSaving] = useState()
+  const [error, setError] = useState(false)
+  const [value1, setValue1] = useState('')
+  const [value2, setValue2] = useState('')
+  const [value3a, setValue3a] = useState('')
+  const [value3b, setValue3b] = useState('')
+  const [value4, setValue4] = useState('')
+  const [value5, setValue5] = useState('')
+  const [value1Error, setValue1Error] = useState(false)
+  const [value2Error, setValue2Error] = useState(false)
+  const [value3Error, setValue3Error] = useState(false)
+  const [value4Error, setValue4Error] = useState(false)
+  const [value5Error, setValue5Error] = useState(false)
```

Comment:
Again, ok for now, but this is an indication that we should add a TODO to decide on a form library for Mapeo to cut down on repeated logic like this. I'm thinking either https://final-form.org/react or https://react-hook-form.com

Reply:
I'll add a TODO for now. I don't have experience with either of these libraries, so I'll await recommendations from the team.

Reply:
I did have some time to look into these briefly, and here's what I've found - the challenge to overcome is how to forward the form library's props to Material UI components.

Someone wrote a nice tutorial about how [to integrate react-hook-form with Material-UI](https://medium.com/@andresss/using-material-ui-with-react-hook-form-is-this-simple-d8383941fafe). It seems react-hook form exports a wrapper that makes this possible, right out of the box.

It seems that there is not a way to do this with React Final Form, so someone [wrote their own wrapper library](https://github.com/lookfirst/mui-rff) to integrate RFF with Material-UI.

That's an edge toward react-hook-form. I also like the syntax on RHF better as it's just hook-based, so everything works within the component itself. RFF on the other hand is component and context-based, so it's a little more complex to use.